src/components/Image/imagePropTypes.js

```diff
@@ -10,7 +10,7 @@ const imagePropTypes = {
     source: PropTypes.oneOfType([
         PropTypes.number,
         PropTypes.shape({
-            uri: PropTypes.string.isRequired,
+            uri: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
```

Comment:
isn't the uri should be always string https://reactnative.dev/docs/image#imagesource?

Reply:
TBH I don't have any suggestions yet, I just found URI is a string in all places for React doc. Also strange that we are using the same pattern [earlier](https://github.com/Expensify/App/commit/d97b3bb6c28c6f8290f725ae257ce36332c85619#diff-2ec3c9975c3041fab68bf23715d93ee40562b28dde1995c364754dae26d01d48L41) and didn't face this issue for android. I will update in case I find something.

Reply:
This will break source prop in other places also.
Please look into this urgently.
cc @parasharrajat

Reply:
You should rather modify the ReceiptUtils.getThumbnailAndImageURIs function ig.

Reply:
@shubham1206agra did you find any issue with it? This thread is just propType definition

Reply:
I think it is just confusing, and should be refrained from doing in my opinion. Cause there's already an option for inputting a number as a source prop, and using a number as URI is counter-intuitive

Reply:
> I think it is just confusing, and should be refrained from doing in my opinion. Cause there's already an option for inputting a number as a source prop, and using a number as URI is counter-intuitive

Agree but if not fixing in one place (root component), we should fix in multiple places (parent components).
Btw both are workarounds. We're already aware of it and going to follow-up better image handling in native.
Since this crash is critical and deploy blocker, we went ahead.

Reply:
<img width="1057" alt="Screenshot 2023-09-04 at 11 21 38 AM" src="https://github.com/Expensify/App/assets/58412969/40c988e5-6706-4f7f-8352-d905a9d7d816">

I think you should change the function here cause it is expecting to receive a string, but it is being supplied a number which will cause problems during TS migration and might also cause unexpected problems

Reply:
And @situchan why does ReportPreview is getting local image object?

Reply:
@situchan I agree with the point. @shubham1206agra here the number is a reference for importing the local images. not `path` as a number.
here you see `ReceiptGenric`  that will import from the local image. so in native when we import a file it will take the refrenece

Reply:
<img width="751" alt="Screenshot 2023-09-04 at 11 35 01 AM" src="https://github.com/Expensify/App/assets/16595846/6f6a4bfd-dbd3-439c-95cf-7711262697d0">
https://reactnative.dev/docs/0.70/image#imagesource

Reply:
Ideally we should refactor `getThumbnailAndImageURIs` function so that it returns full source, not just uri

Reply:
this image passes to  `ReportActionItemImage`

Reply:
Oh, I see. The resolution of this problem is more straightforward than anticipated.
@pradeepmdk @parasharrajat Allow me to fix this in https://github.com/Expensify/App/pull/26452

Reply:
ya because https://github.com/Expensify/App/pull/26452 you are splitting the sourcePropTypes here. so we need to update to allow numbers on native.js.

Reply:
@parasharrajat I also prefer this.